File: src/ProjectR/content/scripts/spells/Anomaly.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Anomaly"
end

function GetDescription()
  return "Deals magic damage to all enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .64
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 3.0625 * aMD - .875 * dMR
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Anomaly.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class Anomaly : SpellScriptBase
    {
        public override string Name { get { return "Anomaly"; } }
        public override string Description { get { return "Deals magic damage to all enemies."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] {  }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 64; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, IList<ICharacter> targets)
        {
            foreach (var target in targets)
            {
                var damage = 3.0625 * MD(caster) - .875 * MR(target);
                DealDamage(target, damage);
            }
        }
    }
}